File: testing/helloMPI.c
#include <stdio.h>
#include <stdlib.h>
#include <mpi.h>
#include "unistd.h"
#include <unistd.h>

int main (int argc, char *argv[])
{
  int rank, size;

  MPI_Init (&argc, &argv);
  MPI_Comm_rank (MPI_COMM_WORLD, &rank);
  MPI_Comm_size (MPI_COMM_WORLD, &size);

  char cwd[1024];
  getcwd(cwd, sizeof(cwd));
  printf( "rank %d of %d working in %s\n", rank, size, cwd);

  MPI_Barrier(MPI_COMM_WORLD);

  MPI_Finalize();

  return 0;
}

File: testing/helloMPI.f90
program hello_world
  implicit none
  include 'mpif.h'
  integer ierr, rank, size
  character*128 cwd

  call MPI_INIT ( ierr )
  call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
  call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)

  call getcwd(cwd)

  write(*,*) "rank ", rank, "of ", size, TRIM(cwd)

  call MPI_BARRIER(MPI_COMM_WORLD, ierr)
 
  call MPI_FINALIZE ( ierr )

end program hello_world
